### PR TITLE
Add sparse gradient option to pretrained embedding

### PR DESCRIPTION
**Pull request by borguz**
Edited clip_grad so that it works with a mix of sparse and dense tensors.
Added sparse argument to pre-trained embedding loading.

### CODE REVIEW

File: torch/nn/modules/sparse.py [hunk 1]
```diff
@@ -122,7 +122,7 @@ def extra_repr(self):
         return s.format(**self.__dict__)
 
     @classmethod
-    def from_pretrained(cls, embeddings, freeze=True):
+    def from_pretrained(cls, embeddings, freeze=True, sparse=False):
```
Comment (soumith) [inline torch/nn/modules/sparse.py]: can you update docstring of the function

File: torch/nn/utils/clip_grad.py [hunk 2]
```diff
@@ -31,7 +31,7 @@ def clip_grad_norm_(parameters, max_norm, norm_type=2):
     clip_coef = max_norm / (total_norm + 1e-6)
     if clip_coef < 1:
         for p in parameters:
-            p.grad.data.mul_(clip_coef.item())
+            p.grad.data.mul_(float(clip_coef))
```
Comment (soumith) [inline torch/nn/utils/clip_grad.py]: this is not a bug. you must have been giving an integer `max_norm`

**Review by soumith — APPROVED** [review]
